eclipse/src/saros/negotiation/additional_resource_data/EclipsePossibleRepresentationProvider.java

```diff
@@ -0,0 +1,34 @@
+package saros.negotiation.additional_resource_data;
+
+import java.util.Collections;
+import java.util.List;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.eclipse.core.resources.IContainer;
+import saros.filesystem.IReferencePoint;
+import saros.filesystem.ResourceConverter;
+import saros.negotiation.AdditionalResourceDataFactory;
+
+/**
+ * Eclipse class providing possible representations of the shared reference point by using the
+ * project name and project-relative path of the reference point delegate.
+ */
+public class EclipsePossibleRepresentationProvider extends AbstractPossibleRepresentationProvider {
+
+  public EclipsePossibleRepresentationProvider(
+      AdditionalResourceDataFactory additionalResourceDataFactory) {
+
+    super(additionalResourceDataFactory);
+  }
+
+  @Override
+  protected List<Pair<String, String>> computePossibleRepresentations(
+      IReferencePoint referencePoint) {
+    IContainer referencePointDelegate = ResourceConverter.getDelegate(referencePoint);
+
+    String projectName = referencePointDelegate.getProject().getName();
+    String projectRelativePath = referencePointDelegate.getProjectRelativePath().toPortableString();
```

Comment:
This is Eclipse specific (the toPortableString() call). I do not see any hints from which IDE this data comes from. Maybe the method does what you need but it is not specified anywhere how a relative should look like.

Can I assume that  the directory Foo\Bar in a Unix environment is correctly invalidated on a Windows machine ?

Reply:
Added javadoc stating that the paths have to use Unix separators. Also adjusts the logic on the receiving side for Eclipse to use `Path.fromPortableString()` to correctly parse the Unix-separated path on Windows.